prost/src/error.rs

```diff
@@ -67,7 +66,71 @@ impl fmt::Display for DecodeError {
         for &(message, field) in &self.inner.stack {
             write!(f, "{message}.{field}: ")?;
         }
-        f.write_str(&self.inner.description)
+        write!(f, "{}", self.inner.description)
+    }
+}
+
+impl From<DecodeErrorKind> for DecodeError {
+    fn from(description: DecodeErrorKind) -> Self {
+        DecodeError {
+            inner: Box::new(Inner {
+                description,
+                stack: Vec::new(),
+            }),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum DecodeErrorKind {
```

Comment:
do we want this to be either `non_exhaustive` or `pub(crate)`? I don't think we want to lock ourselves in like this?

Reply:
You are right, this should be private